pyphare/pyphare_tests/test_pharesee/test_geometry_2d.py

```diff
@@ -0,0 +1,493 @@
+        print("GeometryTest.test_touch_border")
+        hierarchy = super().basic_hierarchy()
+
+        domain_box = hierarchy.domain_box
+        self.assertFalse(touch_domain_border(Box2D(10, 14), domain_box, "upper"))
+        self.assertFalse(touch_domain_border(Box2D(10, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(0, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 14), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 10), domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box2D(-5, 30), domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box2D(10, 30), domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box2D(10, 24), domain_box, "upper"))
+
+
+
+    def test_particle_ghost_area_boxes(self):
+        print("GeometryTest.test_particle_ghost_area_boxes")
+        hierarchy = super().basic_hierarchy()
+
+        expected = {
+            0: [
+                {
+                    "boxes": [
+                        Box([-1, -1], [-1, 10]),
+                        Box([10, -1], [10, 10]),
+                        Box([0, -1], [9, -1]),
+                        Box([0, 10], [9, 10]),
+                    ]
+                },
+                {
+                    "boxes": [
+                        Box([9, 9], [9, 20]),
+                        Box([20, 9], [20, 20]),
+                        Box([10, 9], [19, 9]),
+                        Box([10, 20], [19, 20]),
+                    ]
+                },
+            ],
+            1: [
+                {
+                    "boxes": [  # verify
+                        Box([9, 9], [9, 20]),
+                        Box([20, 9], [20, 20]),
+                        Box([10, 9], [19, 9]),
+                        Box([10, 20], [19, 20]),
+                    ]
+                },
+                {
+                    "boxes": [  # verify
+                        Box([27, 27], [27, 40]),
+                        Box([40, 27], [40, 40]),
+                        Box([28, 27], [39, 27]),
+                        Box([28, 40], [39, 40]),
+                    ]
+                },
+            ],
+        }
+
+        gaboxes = ghost_area_boxes(hierarchy, "particles")
+        particles = "particles"
+
+        self.assertEqual(len(expected), len(gaboxes))
+
+        for ilvl, lvl in enumerate(hierarchy.patch_levels):
+            for act_pdata, exp_pdata in zip(gaboxes[ilvl][particles], expected[ilvl]):
+                self.assertEqual(len(exp_pdata["boxes"]), len(act_pdata["boxes"]))
+                for exp_box in exp_pdata["boxes"]:
+                    self.assertTrue(exp_box in act_pdata["boxes"])
+
+
+
+    @data(
+        (
+            {
+                1: [
+                    {"boxes": [Box([9, 9], [9, 20])]},
+                    {"boxes": [Box([20, 9], [20, 20])]},
+                    {"boxes": [Box([10, 9], [19, 9])]},  # verify
```

Comment:
you can remove the comments.
Note that since the two patches on L1 are not adjacent all their ghost boxes are level ghost boxes therefore the dict is the same as before : should we factor that into one dict to avoid typos and write a small comment in this test to say that all boxes are expected to be lvl ghost since not adjacent?

Reply:
the boxes listed are the level ghost boxes, but I've removed this as it's the same as another test.